src/platform/plugins/shared/discover/public/application/main/state_management/redux/actions/save_discover_session.ts

```diff
@@ -94,6 +95,11 @@ export const saveDiscoverSession = createInternalStateAsyncThunk(
               services,
             })
           );
+          if (newTimeRestore && !updatedTab.timeRange && selectedTab?.globalState.timeRange) {
+            // assign the current time range of the selected tab if time restore is enabled and no time range was set yet for this tab
+            updatedTab.timeRange = selectedTab.globalState.timeRange;
+            updatedTab.refreshInterval = selectedTab.globalState.refreshInterval;
+          }
```

Comment:
Could be a good case to add to the unit tests, even just updating "should include timeRange and refreshInterval when timeRestore is true" would work.